packages/api-graph/src/private/GraphProvider.tsx

```diff
@@ -0,0 +1,105 @@
+import type { DirectLineActivityNode } from '@msinternal/botframework-webchat-core-graph';
+import { reactNode, validateProps } from '@msinternal/botframework-webchat-react-valibot';
+import { createStore, WebChatActivity } from 'botframework-webchat-core';
+import { createGraphFromStore, isOfType, type GraphSubscriber, type Identifier } from 'botframework-webchat-core/graph';
+import React, { memo, useEffect, useMemo, useState } from 'react';
+import { custom, function_, object, optional, parse, pipe, readonly, safeParse, type InferInput } from 'valibot';
+
+import GraphContext, { graphContextSchema, GraphContextType } from './GraphContext';
+
+const EMPTY_ARRAY = Object.freeze([]);
+
+const graphProviderPropsSchema = pipe(
+  object({
+    children: optional(reactNode()),
+    store: custom<ReturnType<typeof createStore>>(
+      value => safeParse(object({ getState: function_(), subscribe: function_() }), value).success
+    )
+  }),
+  readonly()
+);
+
+type GraphProviderProps = InferInput<typeof graphProviderPropsSchema>;
+
+function GraphProvider(props: GraphProviderProps) {
+  const { children, store } = validateProps(graphProviderPropsSchema, props);
+
+  const graph = useMemo(() => createGraphFromStore(store), [store]);
+
+  const [orderedActivityNodes, setOrderedActivityNodes] = useState<readonly DirectLineActivityNode[]>(EMPTY_ARRAY);
+
+  useEffect(() => {
```

Comment:
I imagine this would be [useSyncExternalStore](https://react.dev/reference/react/useSyncExternalStore) in the future which is shimmed for an older React via `useEffect`: https://github.com/facebook/react/blob/main/packages/use-sync-external-store/src/useSyncExternalStoreShimClient.js#L109-L131